mcs/class/test-helpers/PathHelpers.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace MonoTests.Helpers {
+
+	public static class PathHelpers
+	{
+		static string root;
+		static uint last_number;
+
+		static PathHelpers ()
+		{
+			// Create a root temporary directory that is to be the base of all other temporary directories we create.
+			// Contains the assembly name to make the name somewhat intelligible,
+			// and make it include the PID so that parallel execution won't stomp on eachother.
+			root = Path.Combine (Path.GetTempPath (), Assembly.GetExecutingAssembly ().GetName ().Name + "_" + Process.GetCurrentProcess ().Id);
+			// Clean up the directory if it already exists (which would be quite rare, because it would only happen if we happen to run with the same pid at a later moment)
+			DeleteDirectory (root);
+			Directory.CreateDirectory (root);
+
+			// Try to clean up after us when the process exits.
+			AppDomain.CurrentDomain.ProcessExit += Cleanup;
+		}
+
+		static void Cleanup (object sender, EventArgs args)
+		{
+			DeleteDirectory (root);
+		}
+
+		[DllImport ("libc", SetLastError = true)]
+		static extern int mkdir (string path, ushort mode);
+
+		[DllImport ("kernel32.dll", SetLastError = true, CharSet = CharSet.Auto)]
+		static extern bool CreateDirectory (string path, IntPtr securityAttributes);
+
+		static bool RunningOnWindows ()
+		{
+			int i = (int) Environment.OSVersion.Platform;
+			return ((i != 4) && (i != 128));
+		}
+
+		// Tries to recursively delete the specified path.
+		// Doesn't throw exceptions if path is null, empty, or doesn't exist.
+		public static void DeleteDirectory (string path)
+		{
+			if (string.IsNullOrEmpty (path))
+				return;
+
+			if (Directory.Exists (path))
+				Directory.Delete (path, true);
+		}
+
+		// Creates a unique temporary directory.
+		//
+		// The calling method's type/method name will be a part of the
+		// returned path to make it somewhat nicer/more useful.
+		//
+		// This method is only meant for testing code, not production code (it
+		// will leave some temporary directories behind).
+		public static string CreateTemporaryDirectory ()
+		{
+			string name;
+			var calling_method = new StackFrame (1).GetMethod ();
+			if (calling_method != null) {
+				name = calling_method.DeclaringType.FullName + "_" + calling_method.Name;
+			} else {
+				name = "unknown-test";
+			}
+
+			// We store the last number used to create a unique directory, to
```

Comment:
Guid would be much simpler. Possibly removing dashes and braces.

Reply:
Guid would be simpler, but I don't really like filling /tmp with randomly named stuff. It makes me wonder why it's there and who created it (but not enough to actually figure out the why and who, which just leaves me annoyed).

That said, I won't be running the System tests very often, so I'm fine with whatever you think is best.

Reply:
You can prefix the guid with something indicative and constant.